src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,70 @@
+package duke;
+
+import duke.command.Command;
+import duke.storage.SaveHandler;
+import duke.textui.TextUi;
+import duke.usertask.TaskList;
+
+import java.io.IOException;
+
+/**
+ * Duke is a chat-bot program that is capable of logging tasks which are (optionally)
+ * tagged with a deadline/occurrence date.
+ * <p>
+ * Duke automatically saves the tasks of each session upon successful termination
+ * of the session, and restores these tasks on the next launch.
+ */
+public class Duke {
+    private final TaskList tasks;
+    private final TextUi ui;
+    private SaveHandler sv = null;
+
+    /**
+     * Begins execution of the Duke chat-bot.
+     */
+    public void run() {
+        ui.printGreeting();
+        if (sv != null) {
+            try {
+                sv.restore(tasks);
+            } catch (DukeException e) {
+                ui.printFromRed("Oops, something went wrong: " + "\n");
+                ui.printFromRed("** " + e.getMessage() + "\n");
+            }
+        }
+        boolean isRunning = true;
+        while (isRunning) {
+            try {
+                String userInput = ui.awaitInputFromUser();
+                Command cmd = Command.parse(userInput);
+                cmd.execute(ui, tasks);
+                ui.printWithBuffer("\n");
+                isRunning = !cmd.isExit();
+            } catch (DukeException e) {
+                ui.printFromRed("Oops, something went wrong: " + "\n");
+                ui.printFromRed("** " + e.getMessage() + "\n");
+            }
+        }
+        ui.printExitMessage();
+        if (sv != null) {
+            sv.save(tasks);
+        }
+    }
+
+    /**
+     * Creates a new Duke chat-bot instance.
+     */
+    public Duke() {
```

Comment:
Consider moving the constructor to the top

Reply:
Done, and along with all the Javadocs. Thank you for the kind words and feedback!